central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -182,6 +222,10 @@ func (ds *datastoreImpl) AddCollection(ctx context.Context, collection *storage.
 	return err
 }
 
+func (ds *datastoreImpl) DryRunCollection(_ context.Context, collection *storage.ResourceCollection) error {
+	return ds.dryRunCollectionInGraph(collection)
```

Comment:
Collection names are unique. Therefore, you have to validate the name uniqueness in the dry-run. You can hold a set of stored names to perform the name uniqueness validation.

Reply:
There are already unit tests included in this PR for this. The way this gets handled is that the DAG library checks for uniqueness of keys and values, the name field is stored as the value. In the future if we need to certify the uniqueness of more fields we can define a struct to store all the relevant ones

Reply:
Got it!